### PR TITLE
Display template path of current page in dev mode

### PR DESCRIPTION
**Pull request by lunny**
For developing, when a page has problem, we have to investigate which template is the page. Now the template path will be displayed in the footer nearby template load time if it's develop mode.

image

### CODE REVIEW

File: modules/context/context.go [hunk 1]
```diff
@@ -197,6 +197,9 @@ func (ctx *Context) RedirectToFirst(location ...string) {
 func (ctx *Context) HTML(status int, name base.TplName) {
 	log.Debug("Template: %s", name)
 	tmplStartTime := time.Now()
+	if !setting.IsProd {
+		ctx.Data["TmpName"] = name
```
Comment (wxiaoguang) [inline modules/context/context.go]: `TmpName` means `temporary name`

**Comment by wxiaoguang** [comment]
Maybe we can show all rendered template file paths in TRACE level. There are a lot of embedded templates, they can not be found by this PR's mechanism.

**Comment by lunny** [comment]
> Maybe we can show all rendered template file paths in TRACE level. There are a lot of embedded templates, they can not be found by this PR's mechanism.

This will display the entry template path. Once you can open that file, you can find the embedded templates. I don't think it's meaningful to display them in trace log. You can find them in the entry template in any time.

If we really want to know that, we can reject some js or css code to add some box lines and title in the pages to indicate which area is from which template.

File: modules/context/context.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/context/context.go]: Generally LGTM if the variable is clear, eg: `TmplFileName`

**Comment by wxiaoguang** [comment]
> Would suggest `Template <name>: <time>`.

Or maybe: `Page: 279ms Template: 103ms(repo/home)`

Either is fine to me.

**Comment by lunny** [comment]
@wxiaoguang I changed the name and for the position I followed @silverwind's advice because templatename maybe blank. and `:` is one char but  `()` is two. :)

**Review by wxiaoguang — APPROVED** [review]

**Review by singuliere — APPROVED** [review]

File: templates/base/footer_content.tmpl [hunk 2]
```diff
@@ -1,7 +1,7 @@
 <footer>
 	<div class="ui container">
 		<div class="ui left">
-			{{.i18n.Tr "powered_by" "Gitea"}} {{if (or .ShowFooterVersion .PageIsAdmin)}}{{.i18n.Tr "version"}}: {{AppVer}}{{end}} {{if ShowFooterTemplateLoadTime}}{{.i18n.Tr "page"}}: <strong>{{LoadTimes .PageStartTime}}</strong> {{.i18n.Tr "template"}}: <strong>{{call .TmplLoadTimes}}</strong>{{end}}
+			{{.i18n.Tr "powered_by" "Gitea"}} {{if (or .ShowFooterVersion .PageIsAdmin)}}{{.i18n.Tr "version"}}: {{AppVer}}{{end}} {{if ShowFooterTemplateLoadTime}}{{.i18n.Tr "page"}}: <strong>{{LoadTimes .PageStartTime}}</strong> {{.i18n.Tr "template"}} {{if .TemplateName}}{{.TemplateName}}{{end}}: <strong>{{call .TemplateLoadTimes}}</strong>{{end}}
```
Comment (silverwind) [inline templates/base/footer_content.tmpl]:
```suggestion
			{{.i18n.Tr "powered_by" "Gitea"}} {{if (or .ShowFooterVersion .PageIsAdmin)}}{{.i18n.Tr "version"}}: {{AppVer}}{{end}} {{if ShowFooterTemplateLoadTime}}{{.i18n.Tr "page"}}: <strong>{{LoadTimes .PageStartTime}}</strong> {{.i18n.Tr "template"}}{{if .TemplateName}} {{.TemplateName}}{{end}}: <strong>{{call .TemplateLoadTimes}}</strong>{{end}}
```

Better whitespace this way.

**Review by silverwind — APPROVED** [review]

**Pull request merged**